### PR TITLE
- Company form improvements and other adjustments

### LINKED ISSUE #67: Logo upload for Companies
We decided to include company logos for our webpage.

We cannot allow too large images to be displayed on the website, mostly for performance and to reduce bandwidth usage.

Generally:
- [ ] We need to check if company logo exists at all in files. Maybe warn if company has no logo assigned, but throw error if file was specified, but couldn't be found?
- [ ] Check if it's an image file.
- [ ] Check if image has correct size. Should it be constant size [like 300x300], or should they be only min/max in size? A lot of logos are wide, not square.
- [ ] Check file size, because even if image size matches, it could use weird compression, resulting in too large data blob.

### LINKED ISSUE #188: Missing function on Company Show
Right now you can't show company details

image

The list method was moved to the request, so it should be called from this object without any arguments.

### PR DESCRIPTION
**Pull request by sbanasiak**
Pull request mainly covers revamping the company creation form, changes the map library and fixes minor bugs.

### Most important changes:
- The company creation form has been rebuilt
  - The form was divided into three sections: general information, address details and description
  - Added functionality to upload company logos, which closes #67 
  - Added option to generate company coordinates based on the address and adjust them by moving the pin
  - Added Markdown editor to write company description

image
image

- Changed map library from Mapbox to MapLibre
  - Due to outdated dependencies, it was decided to change the map library to Maplibre, which is built on the Mapbox base, so most of the features remain the same.
It is also faster, does not require payment data for registration, and is open source.

- The appearance of the company info box has been changed

image

- Added additional data validations, including image upload, character limit of company description

- Fixed admin functionality for accepting and deleting a company

- Pull request also fixed a bug with the company show and closes #188

### CODE REVIEW

File: environment/dev/php/php.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/sh
+docker exec -it intern-php php $@
```
Comment (kamilpiech97) [inline environment/dev/php/php.sh:2]:
```suggestion
docker exec -it intern-php php $@

```

File: backend/database/factories/CompanyFactory.php [hunk 2]
```diff
@@ -14,19 +15,21 @@
...
 use Internships\Models\User;
-use Smknstd\FakerPicsumImages\FakerPicsumImagesProvider;
+use Internships\Services\LogoGenerator;
 
 class CompanyFactory extends Factory
 {
     public function definition(): array
     {
-        fake()->addProvider(new FakerPicsumImagesProvider(fake()));
+        $name = deep_copy(fake()->company());
```
Comment (agawle) [inline backend/database/factories/CompanyFactory.php:24]: I think deep_copy is unnecessary in that case

**Review by agawle — CHANGES_REQUESTED** [review]

File: backend/database/factories/CompanyFactory.php (hunk 2 above)
Comment (plencka) [reply backend/database/factories/CompanyFactory.php:24]: Ah, that's right. 
It's a leftover from when I was trying to figure out why LogoGenerator was sometimes creating logos with different initials than the company name (which was caused by not setting it up in the configure() method).

File: backend/app/Http/Controllers/CompanyController.php [hunk 3]
```diff
@@ -42,10 +50,8 @@ public function create(): Response
      */
     public function store(CompanyRequest $request): RedirectResponse
     {
-        $request->data();
-
-        return redirect()->route("company-manage")
-            ->with("success", "status.company_created");
+        Session::flash("success", "status.company_created");
+        return Redirect::route("company-show", ["company" => $request->data()]);
```
Comment (kamilpiech97) [inline backend/app/Http/Controllers/CompanyController.php:54]:
```suggestion
        Session::flash("success", "status.company_created");
        
        return Redirect::route("company-show", ["company" => $request->data()]);
```

File: backend/app/Http/Requests/Api/GetCompaniesRequest.php [hunk 4]
```diff
@@ -44,7 +46,15 @@ public function list(
...
     {
+        if (Gate::forUser(auth()->user())->allows(Permission::ManageCompanies->value)) {
+            return Company::query()->orderBy("status", "asc")
+                ->whereNot("status", CompanyStatus::PendingEdited);
+        }
+
         return Company::query()->orderBy("has_signed_papers", "desc")
-            ->where("status", CompanyStatus::Verified);
+            ->where("status", CompanyStatus::Verified)
+            ->when(auth()->user(), function ($query): void {
```
Comment (kamilpiech97) [inline backend/app/Http/Requests/Api/GetCompaniesRequest.php:56]:
```suggestion
            ->when(auth()->user(), function (Builder $query): void {
```
Typing.

**Review by kamilpiech97 — CHANGES_REQUESTED** [review]

**Review by EwelinaLasowy — CHANGES_REQUESTED** [review]
1. When I refresh the page I don't see selected language
image
2. In the console I see a lot of warnings regarding to not translated key, for example:
image

**Review by EwelinaLasowy — CHANGES_REQUESTED** [review]
1. I think that these buttons should be swapped.
image
2. For UX will be better to make confirmation a bit more informative. 
For now we have:
image
And maybe this text: "Usunąć firmę <nazwa>? Nie można tego cofnąć" will be better?
3. What is the difference between routes` / `and `company/manage`? I see the same list of companies.
4. Why navigation is not highlighted when I'm on for example "Strona główna"?
5. I don't like this button in create company form
image and maybe this button "Zgłoś" should be on the last page?

**Comment by sbanasiak** [comment]
> 1. I think that these buttons should be swapped.
>    image
> 2. For UX will be better to make confirmation a bit more informative.
>    For now we have:
>    image
>    And maybe this text: "Usunąć firmę ? Nie można tego cofnąć" will be better?
> 3. What is the difference between routes`/`and `company/manage`? I see the same list of companies.
> 4. Why navigation is not highlighted when I'm on for example "Strona główna"?
> 5. I don't like this button in create company form
>    image and maybe this button "Zgłoś" should be on the last page?

1. Personally, I think the order of the buttons is fine, but I changed apperance of the delete button.
image
2. Done. Now there is an additional information about the inability to undo delete action.
3. Route '/' shows list of companies which we have permission to view, while 'company/manage' shows only companies that we can manage. Company account will only see the companies created by itself, and the administrator will see all companies.
4. Navbar now shows the current sub-page and has been reworked to read items from JS array.
image
5. The appearance of the buttons for switching sections has been changed. The submit button is now in the last section.
image

**Comment by sbanasiak** [comment]
> 1. When I refresh the page I don't see selected language
>    image
> 2. In the console I see a lot of warnings regarding to not translated key, for example:
>    image
> 3. Do we really need to do this in .env file?
>    image

1&2. 
This is a problem that requires more work, so it will be solved in a separate pull request.
3. 
The settings have been removed from .env and a command has been created to make an administrator account: `make:admin {email} {password} {name} {surname}`.

**Pull request merged**